Client/Frontend/Browser/HomePanel/FavoritesViewController.swift

```diff
@@ -161,6 +177,9 @@ class FavoritesViewController: UIViewController, Themeable {
             $0.removeObserver(self, name: .topSitesConversion, object: nil)
             $0.removeObserver(self, name: .privacyModeChanged, object: nil)
         }
+        
+        // Navigating away from NTP counts the current notification as showed.
+        Preferences.NewTabPage.brandedImageShowed.value = true
```

Comment:
There's one architectural caveat to this solution.
When you have a tab with website open and another empty new tab, navigating from NTP to the first tab counts as tab dismissal and the popup is cleared.

No simple way to prevent that, would require a bigger refactor